Sources/SDFCore/MacroInitializerRewriter.swift

```diff
@@ -142,20 +144,64 @@ private final class MacroInitializerRewriter: SyntaxRewriter {
         let infixOperand = InfixOperatorExprSyntax(foldedNode),
         infixOperand.leftOperand.trimmed.description.starts(with: "self.")
       {
-        assignments.append(
-          Assignment(infixOperand.leftOperand.description, statement))
+        let assignment = Assignment(
+          leftOperand: infixOperand.leftOperand.description,
+          item: statement,
+          group: .selfAssignments)
+        groupedAssignments.insert(assignment)
       } else {
-        otherStatements.append(statement)
+        let assignment = Assignment(
+          leftOperand: "",
+          item: statement,
+          group: .otherAssignments)
+        groupedAssignments.insert(assignment)
       }
     }
 
-    let sortedStatements: [CodeBlockItemSyntax] =
-      assignments
-      .sorted { $0.leftOperand < $1.leftOperand }
-      .map { $0.item }
+    let codeBlockItemSyntaxArray = groupedAssignments.toCodeBlockSyntasItemArray()
 
-    let codeBlockListSyntax = CodeBlockItemListSyntax(
-      sortedStatements + otherStatements)
+    let codeBlockListSyntax = CodeBlockItemListSyntax(codeBlockItemSyntaxArray)
     return body.with(\.statements, codeBlockListSyntax)
   }
 }
+
+private struct GroupedAssignments {
+  private var lastInsertedType: AssignmentGroupType = .none
+  private(set) var assignments: [[Assignment]] = []
+
+  mutating func insert(_ assignment: Assignment) {
+    switch lastInsertedType {
+    case .none:
+      assignments.append([assignment])
+    default:
+      let currentArray: [Assignment]
+      if lastInsertedType == assignment.group {
+        currentArray = assignments.removeLast() + [assignment]
```

Comment:
why are we removing last here? it's clearly working given the tests... but I'm struggling to understand what's happening here.

Reply:
I'm trying to append an item to the last array of `assignments`.

Since `removeLast` is O(1) and `append` is also O(1), this is negligible in terms of perf. I agree in terms of code readability is not great though. Any suggestions on how to make it more readable?

Reply:
Ohhh I think I get it now. This makes sense. A comment might help?

Reply:
https://github.com/fdiaz/SafeDI-Formatter/pull/8/files/22c22890a18579fc79d2ad944bf2a1ed4300b927..44fd022e356f6ba1283c70e9e2e4e7f43d8e7549#diff-142e6f5ebc15f3fa17138a4c7c6f14d9aa35e0a512b080815787e9f0e5f079e4